realm/realm-library/src/main/cpp/java_object_accessor.hpp

```diff
@@ -0,0 +1,320 @@
+    util::Any box(float v) const { return _impl::JavaClassGlobalDef::new_float(m_env, v); }
+    util::Any box(int64_t v) const { return _impl::JavaClassGlobalDef::new_long(m_env, v); }
+    util::Any box(util::Optional<bool> v) const { return v ? _impl::JavaClassGlobalDef::new_boolean(m_env, v.value()) : nullptr; }
+    util::Any box(util::Optional<double> v) const { return v ? _impl::JavaClassGlobalDef::new_double(m_env, v.value()) : nullptr; }
+    util::Any box(util::Optional<float> v) const { return v ? _impl::JavaClassGlobalDef::new_float(m_env, v.value()) : nullptr; }
+    util::Any box(util::Optional<int64_t> v) const { return v ? _impl::JavaClassGlobalDef::new_long(m_env, v.value()) : nullptr; }
+    util::Any box(RowExpr) const { REALM_TERMINATE("'RowExpr' not implemented"); }
+
+    // Mixed type is only supported by the Cocoa binding to enable reading
+    // old Realm files that may have used them. All other bindings can ignore it.
+    util::Any box(Mixed) const { REALM_TERMINATE("'Mixed' not supported"); }
+
+    // Convert from the boxed type to core types. This needs to be implemented
+    // for all of the types which `box()` can take, plus `RowExpr` and optional
+    // versions of the numeric types, minus `List` and `Results`.
+    //
+    // `create` and `update` are only applicable to `unbox<RowExpr>`. If
+    // `create` is false then when given something which is not a managed Realm
+    // object `unbox()` should simply return a detached row expr, while if it's
+    // true then `unbox()` should create a new object in the context's Realm
+    // using the provided value. If `update` is true then upsert semantics
+    // should be used for this.
+    template<typename T>
+    T unbox(util::Any& /*v*/, bool /*create*/= false, bool /*update*/= false) const {
+        throw std::logic_error("Missing template specialization"); // All types should have specialized templates
+    }
+
+    bool is_null(util::Any const& v) const noexcept { return !v.has_value(); }
+    util::Any null_value() const noexcept { return {}; }
+    util::Optional<util::Any> no_value() const noexcept { return {}; }
+
+    // KVO hooks which will be called before and after modying a property from
+    // within Object::create().
+    void will_change(Object const&, Property const&) {}
+    void did_change() {}
+
+    // Get a string representation of the given value for use in error messages.
+    // This method should only be used when printing warnings about primary keys
+    // which means the input should only be valid types for primary keys:
+    // StringData, int64_t and Optional<int64_t>
+    std::string print(util::Any const& val) const {
+        if (!val.has_value()) {
+            return "null";
+        }
+
+        if (val.type() == typeid(StringData)) {
+            auto str = any_cast<StringData>(val);
+            if (str.is_null()) {
+                return "null";
```

Comment:
Consider `(null)` instead.

Reply:
This method should return the string representation of whatever values is parsed in. I'm not sure I follow why `(null)` is better since I assume it involves some automatic conversion somewhere?

Reply:
Good point. Right now we wrap all values in `''` which makes it hard to see what is strings and what is not.